internal/core/software_renderer.rs

```diff
@@ -1667,6 +1731,18 @@ impl<'a, T: ProcessScene> crate::item_rendering::ItemRenderer for SceneBuilder<'
                 self.draw_text_paragraph(paragraph, physical_clip, offset, color);
             }
         }
+
+        let cursor_rect = self
+            .window
+            .window_adapter()
+            .renderer()
+            .text_input_cursor_rect_for_byte_offset(
+                text_input,
+                text_input.cursor_position_byte_offset().round() as usize,
+            )
+            .cast_unit::<crate::lengths::PhysicalPx>();
```

Comment:
Same issue with the cast here, `text_input_cursor_rect_for_byte_offset` returns a rect in logical coordinates, so it's wrong to just cast it to the physical coordinate system without applying the scale factor.